### PR TITLE
docs: PCP-3141

### PR DESCRIPTION
**Pull request by karl-cardenas-coding**
## Describe the Change

This PR documents the ability for Azure clusters to honor network proxy configurations that may be required for outbound network traffic. The end user must meet a couple of requirements to use this feature. The [Architecture Overview](https://deploy-preview-3378--docs-spectrocloud.netlify.app/clusters/public-cloud/azure/architecture/#proxy-configuration) explains at a high-level what is needed and links to the various resources required. 

The pages listed below were updated to help consumers find the information more easily.

## Changed Pages

💻 [Architecture Overview](https://deploy-preview-3378--docs-spectrocloud.netlify.app/clusters/public-cloud/azure/architecture/#proxy-configuration)
- [Enable and Manage Proxy Configurations](https://deploy-preview-3378--docs-spectrocloud.netlify.app/clusters/pcg/manage-pcg/configure-proxy/)
- [Manage PGC Index](https://deploy-preview-3378--docs-spectrocloud.netlify.app/clusters/pcg/manage-pcg/)
- [Register and Manage Azure Cloud Account](https://deploy-preview-3378--docs-spectrocloud.netlify.app/clusters/public-cloud/azure/azure-cloud/)
- [Create and Manage Azure IaaS Cluster](https://deploy-preview-3378--docs-spectrocloud.netlify.app/clusters/public-cloud/azure/create-azure-cluster/)
- [Create and Manage Azure AKS Cluster](https://deploy-preview-3378--docs-spectrocloud.netlify.app/clusters/public-cloud/azure/aks/)
- [Release Notes](https://deploy-preview-3378--docs-spectrocloud.netlify.app/release-notes/#features)

## Jira Tickets

🎫 [PCP-3141](https://spectrocloud.atlassian.net/browse/PCP-3141)

## Backports

Can this PR be backported?

- [ ] Yes. _Remember to add the relevant backport labels to your PR._
- [x] No. _Please leave a short comment below about why this PR cannot be backported._

[PCP-3141]: https://spectrocloud.atlassian.net/browse/PCP-3141?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/docs-content/clusters/pcg/manage-pcg/configure-proxy.md [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+  manage the proxy configurations.
+
+## Enable Proxy
+
+1. Open a terminal session and navigate to the folder you have the Reach Helm Chart zip file downloaded to.
+
+2. Unzip the downloaded artifact you received from the support team.
+
+   ```shell
+   unzip release-*.zip -d palette
```
[the remark this answers is not in the document]
Comment (karl-cardenas-coding) [reply docs/docs-content/clusters/pcg/manage-pcg/configure-proxy.md:87]: This one has no longform 😢

File: docs/docs-content/clusters/pcg/manage-pcg/configure-proxy.md [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+workload clusters. Individual workload clusters must have their proxy configurations to access the internet.
+
+:::
+
+Use the following steps to add and manage proxy configurations for a PCG.
+
+## Prerequisites
+
+- A PCG is deployed into an active Kubernetes cluster, which is active and healthy. Refer to
+  [Deploy a PCG to an Existing Kubernetes Cluster](../deploy-pcg-k8s.md) for additional guidance.
```
Comment (yuliiiah) [inline docs/docs-content/clusters/pcg/manage-pcg/configure-proxy.md:31]:
```suggestion
- A PCG is deployed into an active and healthy Kubernetes cluster. Refer to
  [Deploy a PCG to an Existing Kubernetes Cluster](../deploy-pcg-k8s.md) for additional guidance.
```

File: docs/docs-content/clusters/pcg/manage-pcg/configure-proxy.md [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+
+  :::warning
+
+  If you deployed a [PCG through the Palette CLI](../pcg.md#supported-environments), refer to the respective platform
+  install guide for instructions on how to configure proxy settings during the installation process through the CLI.
```
Comment (yuliiiah) [inline docs/docs-content/clusters/pcg/manage-pcg/configure-proxy.md:36]:
```suggestion
  If you deployed a [PCG through the Palette CLI](../pcg.md#supported-environments), refer to the respective platform
  installation guide for instructions on how to configure proxy settings during the installation process through the CLI.
```

File: docs/docs-content/clusters/pcg/manage-pcg/configure-proxy.md [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+
+  :::
+
+- The kubeconfig file for the Kubernetes cluster where the PCG is deployed. The Kubconfig file is used to authenticate
+  with the Kubernetes cluster and deploy the Reach service.
+
+- Admin access to the Kubernetes cluster where the PCG is deployed. The Reach Helm Chart will create a namespace,
+  service accounts, and roles in the cluster.
+
+- The extract utilities `zip` and `tar` are installed on the machine you are using to deploy the Helm chart.
```
Comment (yuliiiah) [inline docs/docs-content/clusters/pcg/manage-pcg/configure-proxy.md:46]:
```suggestion
- The extract utilities `zip` and `tar` are installed in the system you are using to deploy the Helm chart.
```

Comment (yuliiiah) [reply docs/docs-content/clusters/pcg/manage-pcg/configure-proxy.md:46]: I see "machine" as jargon and prefer to replace it with "system" 👁️

File: docs/docs-content/clusters/pcg/manage-pcg/configure-proxy.md [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+
+- Tenant administrator access.
```
Comment (yuliiiah) [inline docs/docs-content/clusters/pcg/manage-pcg/configure-proxy.md:48]: In Palette?

File: docs/docs-content/clusters/pcg/manage-pcg/configure-proxy.md [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+
+- Proxy configuration details, such as the proxy URL, port, and authentication credentials.
+
+- The deployed PCG must have network connectivity to the proxy server.
+
+- The Kubernetes cluster where the PCG is deployed must have the proxy configuration. This includes any Certificate
+  Authority (CA) certificates are required to authenticate the proxy server. This step varies depending on the platform
```
Comment (yuliiiah) [inline docs/docs-content/clusters/pcg/manage-pcg/configure-proxy.md:55]:
```suggestion
  Authority (CA) certificates that are required to authenticate with the proxy server. This step varies depending on the platform
```

File: docs/docs-content/clusters/pcg/manage-pcg/configure-proxy.md [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+    :::
+
+- Download the Reach Helm Chart provided by our support team. The Reach Helm Chart is used to deploy the Reach service
+  into the Kubernetes cluster where the PCG is deployed. The Reach service is used to manage proxy configurations for
+  the PCG. Contact our support team to obtain the Reach Helm Chart.
+
+- `cert-manager` is not already deployed in the Kubernetes cluster where the PCG is deployed. If `cert-manager` is
+  already deployed, make sure you edit the deployment to use the
+  `- --feature-gates=AdditionalCertificateOutputFormats=true` flag. Otherwise, the Reach service cannot manage the proxy
+  configurations.
```
Comment (yuliiiah) [inline docs/docs-content/clusters/pcg/manage-pcg/configure-proxy.md:77]: In my view, this doesn't fit the structure of a prerequisite. It's a conditional statement, and so I would rather add this as an admonition in the guide and rephrase it to say smth like "If `cert-manager` is already installed, use the following flat..."

File: docs/docs-content/clusters/pcg/manage-pcg/configure-proxy.md [hunk 8]
```diff
@@ -0,0 +1,208 @@
...
+
+## Enable Proxy
+
+1. Open a terminal session and navigate to the folder where you downloaded the Reach Helm Chart zip file.
+
+2. Unzip the downloaded artifact you received from the support team.
```
Comment (yuliiiah) [inline docs/docs-content/clusters/pcg/manage-pcg/configure-proxy.md:83]: Are you referring to the Reach helm zip? I find the word "artifact" to be confusing here, since we're talking about a specific file in the previous step and call out an ambiguous artifact in the next 🤔

File: docs/docs-content/clusters/pcg/manage-pcg/configure-proxy.md [hunk 9]
```diff
@@ -0,0 +1,208 @@
...
+
+   ```shell
+   unzip release-*.zip -d palette
+   ```
+
+3. Navigate to the release folder inside the unzipped folder.
```
Comment (yuliiiah) [inline docs/docs-content/clusters/pcg/manage-pcg/configure-proxy.md:89]:
```suggestion
3. Navigate to the release folder.
```

Comment (yuliiiah) [reply docs/docs-content/clusters/pcg/manage-pcg/configure-proxy.md:89]: I don't think we need to specify "the unzipped folder" if this step follows directly from the previous one 🤔

File: docs/docs-content/clusters/pcg/manage-pcg/configure-proxy.md [hunk 10]
```diff
@@ -0,0 +1,208 @@
...
+   cert-manager extras/cert-manager/cert-manager-*.tgz --install
+   ```
+
+5. Next, navigate to the reach-system folder and extract the Reach Helm Chart.
+
+   ```shell
+   cd extras/reach-system/ && tar -xvzf reach-system-*.tgz
+   ```
+
+6. Use a text editor and open the **values.yaml** file that is inside the **reach-system** directory. Fill out the
```
Comment (yuliiiah) [inline docs/docs-content/clusters/pcg/manage-pcg/configure-proxy.md:109]:
```suggestion
6. Open the **reach-system/values.yaml** file in a text editor. Fill out the
```

File: docs/docs-content/clusters/pcg/manage-pcg/configure-proxy.md [hunk 11]
```diff
@@ -0,0 +1,208 @@
...
+   following YAML fields with the proxy configuration details:
+
+   - `reachSystem.enabled`: Set this field to `true` to enable the Reach service.
+   - `reachSystem.proxySettings.http_proxy`: The HTTP proxy URL, including the port number.
+   - `reachSystem.proxySettings.https_proxy`: The HTTPS proxy URL, including the port number.
+   - `reachSystem.proxySettings.no_proxy`: A comma-separated list of URLs that should bypass the proxy.
+   - `reachSystem.proxySettings.ca_crt_path`: The path to the CA certificate file used to authenticate the proxy server.
+     Make sure the CA is in PEM format. If you do not have a CA certificate, leave this field empty.
```
Comment (yuliiiah) [inline docs/docs-content/clusters/pcg/manage-pcg/configure-proxy.md:117]: I think this should be a table, to stay consistent with the other instances where we instruct users to fill out YAMLs 🤔

Comment (yuliiiah) [inline docs/docs-content/clusters/pcg/manage-pcg/configure-proxy.md:117]:
```suggestion
     Make sure the CA certificate is in the PEM format. If you do not have a CA certificate, leave this field empty.
```

File: docs/docs-content/clusters/pcg/manage-pcg/configure-proxy.md [hunk 12]
```diff
@@ -0,0 +1,208 @@
...
+through Palette.
+
+## Validate
+
+Use the following steps to validate that the Reach service is deployed and that the proxy configuration is enabled for
+the PCG.
+
+1. Open a terminal session.
+
+2. Issue the following command against the Kubernetes cluster where the PCG is deployed. Verify the pods are active and
```
Comment (yuliiiah) [inline docs/docs-content/clusters/pcg/manage-pcg/configure-proxy.md:178]:
```suggestion
2. Issue the following command against the Kubernetes cluster where the PCG is deployed. Verify that the pods are active and
```

File: docs/docs-content/clusters/pcg/manage-pcg/configure-proxy.md [hunk 13]
```diff
@@ -0,0 +1,208 @@
...
+   healthy.
+
+   ```shell
+   kubectl get pods --namespace reach-system
+   ```
+
+   The output should display the Reach service pods in a healthy state.
```
Comment (yuliiiah) [inline docs/docs-content/clusters/pcg/manage-pcg/configure-proxy.md:185]:
```suggestion
   The output should display the Reach service pods in the **Running** status.
```

Comment (yuliiiah) [reply docs/docs-content/clusters/pcg/manage-pcg/configure-proxy.md:185]: I feel like calling out the status relates to the example output more closely 👀

File: docs/docs-content/clusters/pcg/manage-pcg/configure-proxy.md [hunk 14]
```diff
@@ -0,0 +1,208 @@
...
+
+   ```shell hideClipboard
+   NAME                                                READY   STATUS    RESTARTS   AGE
+   reach-controller-manager-995c74db5-frb5j            2/2     Running   0          2m
+   ```
+
+3. After you configure your cloud account to use the PCG. You can verify that deployed clusters inherit the proxy
```
Comment (yuliiiah) [inline docs/docs-content/clusters/pcg/manage-pcg/configure-proxy.md:192]:
```suggestion
3. After configuring your cloud account to use the PCG, you can verify that the deployed workload clusters inherit the proxy
```

File: docs/docs-content/clusters/public-cloud/azure/aks.md [hunk 15]
```diff
@@ -164,16 +164,16 @@ explains how you can create an Azure AKS cluster managed by Palette.
...
+    | **Static Placement**       | By default, Palette uses dynamic placement. This creates a new VNet for clusters with two subnets in different Availability Zones (AZs). Palette places resources in these clusters, manages the resources, and deletes them when the corresponding cluster is deleted.<br /><br />If you want to place resources into a pre-existing VNet, enable the **Static Placement** option and fill out the input values listed in the [Static Placement](#static-placement-settings) table below. <br /> <br /> Select **Static Placement** for clusters that you want to use your network proxy configurations. To learn more about proxy configurations, check out [Proxy Configuration](./architecture.md#proxy-configuration). |
```
Comment (yuliiiah) [inline docs/docs-content/clusters/public-cloud/azure/aks.md:176]:
```suggestion
    | **Static Placement**       | By default, Palette uses dynamic placement. This creates a new VNet for clusters with two subnets in different Availability Zones (AZs). Palette places resources in these clusters, manages the resources, and deletes them when the corresponding cluster is deleted.<br /><br />If you want to place resources into a pre-existing VNet, enable the **Static Placement** option and fill out the input values listed in the [Static Placement](#static-placement-settings) table below. <br /> <br /> Select **Static Placement** for clusters where you want to use your network proxy configurations. To learn more about proxy configurations, check out [Proxy Configuration](./architecture.md#proxy-configuration). |
```

File: docs/docs-content/clusters/public-cloud/azure/architecture.md [hunk 16]
```diff
@@ -165,3 +165,36 @@ you use any of the following tags.
 - `microsoft`
 
 - `windows`
+
+## Proxy Configuration
+
+You can enable your Azure clusters to use a proxy server for outbound traffic. To use your proxy server with Azure
+clusters, you must deploy a Private Cloud Gateway (PCG) in your Azure environment. The PCG must be configured with the
+proxy server details. Once the PCG is deployed and configured with the proxy server details, newly deployed Azure
```
Comment (yuliiiah) [inline docs/docs-content/clusters/public-cloud/azure/architecture.md:173]:
```suggestion
proxy server details. Once the PCG is deployed and configured with the proxy server details, the newly deployed Azure
```

File: docs/docs-content/clusters/public-cloud/azure/architecture.md [hunk 17]
```diff
@@ -165,3 +165,36 @@ you use any of the following tags.
...
+clusters will inherit the proxy configurations from the PCG.
+
+:::tip
+
+We recommend you review the [gRPC and Proxies](../../../architecture/grps-proxy.md) to be aware of network proxies
```
Comment (yuliiiah) [inline docs/docs-content/clusters/public-cloud/azure/architecture.md:178]:
```suggestion
We recommend you review the [gRPC and Proxies](../../../architecture/grps-proxy.md) to be aware of network proxies that
```

File: docs/docs-content/clusters/public-cloud/azure/architecture.md [hunk 18]
```diff
@@ -165,3 +165,36 @@ you use any of the following tags.
...
+2. Configure the PCG with the proxy server details. For instructions on how to do this, refer to the
+   [Enable and Manage Proxy Configurations](../../pcg/manage-pcg/configure-proxy.md) guide.
+
+3. Configure your Azure account to use the PCG. When adding an Azure account to Palette, select the **Connect Private
+   Cloud Gateway** option and select the PCG cluster you deployed. Refer to the
+   [Add Azure Cloud Account](./azure-cloud.md) guide for instructions on how to do this.
+
+4. Deploy an Azure cluster and use static placement so that you can specify the VNet with the proper network
+   configuration. If you do not use static placement, the Azure cluster will be deployed with the network resources
+   created by Palette, which may not have the proper network configuration. For instructions on how to deploy an Azure
```
Comment (yuliiiah) [inline docs/docs-content/clusters/public-cloud/azure/architecture.md:198]: > If you do not use static placement, the Azure cluster will be deployed with the network resources
   created by Palette, which may not have the proper network configuration.

I feel like this would better serve as a warning 🤔

File: docs/docs-content/clusters/public-cloud/azure/azure-cloud.md [hunk 19]
```diff
@@ -47,6 +47,14 @@ Use the following steps to add an Azure or Azure Government account in Palette o
 | **Disable Properties**            | This option prevents Palette and VerteX from creating Azure Virtual Networks (VNets) and other network resources on your behalf for static placement deployments. If you enable this option, all users must manually specify a pre-existing VNet, subnets, and security groups when creating clusters. |
 | **Connect Private Cloud Gateway** | Select this option to connect to a Private Cloud Gateway (PCG) if you have a PCG deployed in your environment. Refer to the PCG [Architecture](../../pcg/architecture.md) page to learn more about a PCG.                                                                                              |
 
+:::info
```
Comment (yuliiiah) [inline docs/docs-content/clusters/public-cloud/azure/azure-cloud.md:50]: This admonition and the table above it are not properly aligned with step five 👁️

File: docs/docs-content/clusters/public-cloud/azure/azure-cloud.md [hunk 20]
```diff
@@ -47,6 +47,14 @@ Use the following steps to add an Azure or Azure Government account in Palette o
...
+
+If you want to provide network proxy configurations to your Kubernetes clusters deployed through Palette, you must use a
```
Comment (yuliiiah) [inline docs/docs-content/clusters/public-cloud/azure/azure-cloud.md:52]:
```suggestion
If you want to provide network proxy configurations to the Kubernetes clusters deployed through Palette, you must use a
```

File: docs/docs-content/clusters/public-cloud/azure/create-azure-cluster.md [hunk 21]
```diff
@@ -140,15 +140,15 @@ Use the following steps to deploy an Azure cluster.
...
+    | **Static Placement**  | By default, Palette uses dynamic placement. This creates a new VNet for clusters with two subnets in different Availability Zones (AZs). Palette places resources in these clusters, manages the resources, and deletes them when the corresponding cluster is deleted.<br /><br />If you want to place resources into a pre-existing VNet, enable the **Static Placement** option, and fill out the input values listed in the [Static Placement](#static-placement-settings) table below. <br /> <br /> Select **Static Placement** for clusters that you want to use your network proxy configurations. To learn more about proxy configurations, check out [Proxy Configuration](./architecture.md#proxy-configuration). |
```
Comment (yuliiiah) [inline docs/docs-content/clusters/public-cloud/azure/create-azure-cluster.md:151]:
```suggestion
    | **Static Placement**  | By default, Palette uses dynamic placement. This creates a new VNet for clusters with two subnets in different Availability Zones (AZs). Palette places resources in these clusters, manages the resources, and deletes them when the corresponding cluster is deleted.<br /><br />If you want to place resources into a pre-existing VNet, enable the **Static Placement** option, and fill out the input values listed in the [Static Placement](#static-placement-settings) table below. <br /> <br /> Select **Static Placement** for clusters where you want to use your network proxy configurations. To learn more about proxy configurations, check out [Proxy Configuration](./architecture.md#proxy-configuration). |
```

File: docs/docs-content/release-notes/release-notes.md [hunk 22]
```diff
@@ -24,6 +24,12 @@ tags: ["release-notes"]
   the feature. Check out the [Feature Flags](../enterprise-version/system-management/feature-flags.md) documentation to
   learn more.
 
+- Kubernetes clusters deployed on Azure can now use network proxy configurations. To use this new feature, you must
```
Comment (yuliiiah) [inline docs/docs-content/release-notes/release-notes.md:27]:
```suggestion
- Kubernetes clusters deployed to Azure can now use network proxy configurations. To use this new feature, you must
```

File: docs/docs-content/release-notes/release-notes.md [hunk 23]
```diff
@@ -24,6 +24,12 @@ tags: ["release-notes"]
...
+  deploy a PCG in your Azure environment and configure the PCG to use your network proxy server. Once the PCG is
+  deployed and configured with the proxy server details, newly deployed Azure clusters will inherit the proxy
```
Comment (yuliiiah) [inline docs/docs-content/release-notes/release-notes.md:29]:
```suggestion
  deployed and configured with the proxy server details, the newly deployed Azure clusters will inherit the proxy
```

**Review by yuliiiah — APPROVED** [review]
Approved with some questions, copy edits, and suggestions ✨

File: docs/docs-content/clusters/pcg/deploy-pcg-k8s.md [hunk 24]
```diff
@@ -114,6 +114,14 @@ Use the following steps to install the PCG in your existing Kubernetes cluster.
    field will display **Pending** while the PCG is deploying. The deployment is complete when the **Cluster Status**
    field displays the status **Running**.
 
+   :::tip
+
+   You can provide network proxy configurations to your Kubernetes clusters deployed through Palette. To provide network
+   proxy configurations to your host clusters, update the PCG with the proxy server details. To learn more, check out
+   the [Proxy Configuration](./manage-pcg/configure-proxy.md) guide.
```
[the remark this answers is not in the document]
Comment (karl-cardenas-coding) [reply docs/docs-content/clusters/pcg/deploy-pcg-k8s.md:121]: We use both.

File: docs/docs-content/clusters/pcg/manage-pcg/configure-proxy.md (hunk 5 above)
Comment (karl-cardenas-coding) [inline docs/docs-content/clusters/pcg/manage-pcg/configure-proxy.md:48]:
```suggestion
- Palette tenant administrator access.
```

Comment (karl-cardenas-coding) [reply docs/docs-content/clusters/pcg/manage-pcg/configure-proxy.md:48]: Good callout :)

File: docs/docs-content/clusters/pcg/manage-pcg/configure-proxy.md (hunk 8 above)
Comment (karl-cardenas-coding) [inline docs/docs-content/clusters/pcg/manage-pcg/configure-proxy.md:83]:
```suggestion
2. Unzip the zip file you received from the support team.
```

File: docs/docs-content/clusters/pcg/manage-pcg/configure-proxy.md (hunk 7 above)
Comment (karl-cardenas-coding) [reply docs/docs-content/clusters/pcg/manage-pcg/configure-proxy.md:79]: Updated to use admonition.

File: docs/docs-content/clusters/public-cloud/azure/architecture.md (hunk 18 above)
Comment (karl-cardenas-coding) [reply docs/docs-content/clusters/public-cloud/azure/architecture.md:198]: Done. Good callout.

File: docs/docs-content/clusters/public-cloud/azure/azure-cloud.md (hunk 19 above)
Comment (karl-cardenas-coding) [reply docs/docs-content/clusters/public-cloud/azure/azure-cloud.md:50]: Eagle eye 🦅
